docs/content/docs/security/ssl.md

```diff
@@ -0,0 +1,367 @@
+
+There are two types of network connections to authenticate and encrypt: internal and external.
+
+{{< img src="/fig/ssl_internal_external.svg" alt="Internal and External Connectivity" width=75% >}}
+
+For more flexibility, security for internal and external connectivity can be enabled and configured
+separately.
+
+### Internal Connectivity
+
+Flink internal communication refers to all connections made between Flink processes. These include:
+
+- Control messages: RPC between JobManager / TaskManager / Dispatcher / ResourceManager
+- Transfers on the data plane: connections between TaskManagers to exchange data during shuffles, 
+  broadcasts, redistribution, etc
+- Blob service communication: distribution of libraries and other artifacts
+
+All internal connections are SSL authenticated and encrypted. The connections use **mutual authentication**,
+meaning both server and client side of each connection need to present the certificate to each other. 
+The certificate acts as a shared secret and can be embedded into container images or attached to your 
+deployment setup. These connections run Flink custom protocols. Users never connect directly to internal 
+connectivity endpoints.
+
+### External Connectivity
+
+Flink external communication refers to all connections made from the outside to Flink processes. 
+This includes: 
+- communication with the Dispatcher to submit Flink jobs (session clusters)
+- communication of the Flink CLI with the JobManager to inspect and modify a running Flink job/application
+
+These connections are exposed via REST/HTTP endpoints (and are used by Flink's web UI). Some 
+application-specific external services such as sources and sinks may use some other network protocol.
+
+The server will, by default, accept connections from any client, meaning that the REST endpoint does 
+not authenticate the client. These REST endpoints, however, can be configured to require SSL encryption 
+and mutual authentication. 
+
+However, the recommended approach is setting up and configuring a dedicated proxy service (a "sidecar 
+proxy") that controls access to the REST endpoint. This involves binding the REST endpoint to the 
+loopback interface (or the pod-local interface in Kubernetes) and starting a REST proxy that authenticates 
+and forwards the requests to Flink. Examples for proxies that Flink users have deployed are [Envoy Proxy](https://www.envoyproxy.io/) 
+or [NGINX with MOD_AUTH](http://nginx.org/en/docs/http/ngx_http_auth_request_module.html).
+
+The rationale behind delegating authentication to a proxy is that such proxies offer a wide variety
+of authentication options and thus better integration into existing infrastructures.
+
+## Queryable State
+
+Connections to the [queryable state]({{< ref "docs/dev/datastream/fault-tolerance/queryable_state" >}}) 
+endpoints is currently not authenticated or encrypted.
+
+## SSL Setups
+
+{{< img src="/fig/ssl_mutual_auth.svg" alt="SSL Mutual Authentication" width=75% >}}
```

Comment:
This is still not working. Not sure what is the issue but having such issue is something we need to fix I think.
<img width="427" alt="Screenshot 2022-02-16 at 12 51 40" src="https://user-images.githubusercontent.com/18561820/154259238-2e7e4f58-5a1a-4083-9e7e-a3590ce8a088.png">

Reply:
@infoverload I've taken a deeper look and found the issue. The mentioned `svg` file has `html` content. Please save the proper `svg` file into `ssl_mutual_auth.svg` and then everything will work like charm.